### DISCUSSION TITLE
Replace dynamic values in content

### DISCUSSION THREAD

**Discussion by yandodov** [payloadcms/payload#1810] [Q&A]
I want to implement dynamic values in Payload to add support for text like:

> We currently have `{{ users_count }}` number of users.

From my understanding, I have to use the [`afterRead` collection hook](https://payloadcms.com/docs/hooks/collections#afterread). However, I need some sort of check to see if the read comes from the admin or not.

If you're editing the field in the admin, the value **should not** be substituted, so that you can edit the variable. However, if you directly call the REST or GraphQL API from the front-end, the value **should** be substituted.

How do I check if the value comes from the admin? Alternatively, is there another way to implement this altogether?

**Comment by JarrodMFlesch** [comment]
Hey @hdodov I would love to help here, I think there are a couple ways to solve this. 

1. Create a new `readOnly` field, use the afterRead hook to populate the value and use that on the frontend
2. Create a virtual field, the afterRead hook would add the virtual field to the response (only issue is you will have to adjust types on the frontend, since this field will not live in the config)
3. I believe you could check the req.host (assuming cms and frontend are split on subdomains) to see where the request was made from, if it was from the frontend domain, generate the string, else use the value

Am I understanding your question correctly? Let me know!

**Reply by yandodov** [reply]
I actually needed to be able to manage those variables in the admin, so I created a separate collection for them. Then, I used an afterRead hook to substitute whenever a `?replace=1` parameter is given. I decided that's the best way. On my front-end, I'm going to be using something like Axios, so adding this parameter to all requests should be trivial. The benefit is that I have full control over it.

Here's a gist: https://gist.github.com/hdodov/bbb2e380aa587a100272b298e12e493a

If anyone else stumbles upon this and needs a functionality like that, I could create a plugin.

**Comment by christian-reichart** [comment]
You can check the API via

`req.payloadAPI !== 'local'`

**Reply by yandodov** [reply]
I think I tried that, but it was always `"local"`.
